Api/NodeRepositoryInterface.php

```diff
@@ -6,15 +6,42 @@
 
 interface NodeRepositoryInterface
 {
+    /**
+     * @param \Snowdog\Menu\Api\Data\NodeInterface $page
+     * @return \Snowdog\Menu\Api\Data\NodeInterface
+     */
     public function save(NodeInterface $page);
 
+    /**
+     * @param int $id
+     * @return \Snowdog\Menu\Model\Menu\Node
+     */
     public function getById($id);
 
+    /**
+     * @param \Magento\Framework\Api\SearchCriteriaInterface $criteria
+     * @return \Magento\Framework\Api\SearchResultsInterface
+     */
     public function getList(SearchCriteriaInterface $criteria);
 
+    /**
+     * @param \Snowdog\Menu\Api\Data\NodeInterface $page
+     * @return bool
+     */
     public function delete(NodeInterface $page);
 
+    /**
+     * @param int $id
+     * @return bool
+     */
     public function deleteById($id);
 
-    public function getByMenu($menuId);
+    /**
+     * Return node by menu id
+     *
+     * @api
+     * @param int $menuId
+     * @return \Snowdog\Menu\Api\Data\NodeInterface[]
+     */
+    public function getByMenuId($menuId);
```

Comment:
This change should be documented in changelog because BC breaks. Also why not use real typehinting for `int`?

Reply:
Yep, we shouldn't rename this method without a good reason.

We don't want to use type hints yet, because we have a few Magento 2.1 projects and I know there are issues with that (for example https://github.com/magento/magento2/issues/7620).

Reply:
function name reverted